### PR TITLE
libc: stat: enable fifo_type test case

### PR DESCRIPTION
**Pull request by mateusz-bloch**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by damianloew — APPROVED** [review]

File: libc/misc/stat.c [hunk 1]
```diff
@@ -500,9 +501,12 @@ TEST(stat_mode, fifo_type)
 	TEST_ASSERT_EQUAL_INT(0, lstat(tempPath, &buffer));
 	TEST_ASSERT_TRUE((buffer.st_mode & S_IFMT) == S_IFIFO);
 
+	fifo_fd = open(tempPath, O_NONBLOCK);
```
Comment (damianloew) [inline libc/misc/stat.c:504]: Is it done 2 times?

**Review by damianloew — CHANGES_REQUESTED** [review]

File: libc/misc/stat.c (hunk 1 above)
Comment (mateusz-bloch) [reply libc/misc/stat.c:504]: One for fifo created with 777 permissions, and second time for fifo with 000 permissions

Comment (damianloew) [reply libc/misc/stat.c:504]: Right, it wasn't visible by dafault while reviewing, LGMT

**Review by damianloew — APPROVED** [review]

**Comment by gerard5** [comment]
@mateusz-bloch  according to https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1013 the tests of `mkfifo()` on `STM32L4-nucleo` target `can now be enabled`. The `posixsrv` is now available on this target, please verify if it works for you.

**Pull request merged**